Domain/Repositories/BookingRepository.cs

```diff
@@ -0,0 +1,157 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.IO;
+
+
+using AirportTicketBookingSystem.Domain.Models;
+using AirportTicketBookingSystem.Domain.Services;
+using System.Diagnostics;
+
+namespace AirportTicketBookingSystem.Domain.Repositories
+{
+    public class BookingRepository
+    {
+        private readonly string _filePath;
+        private readonly List<Booking> _bookings;
+        private readonly FlightService _flightsService;
+
+        public BookingRepository()
+        {
+            string relativePath = @"Data\bookings.csv";
+            _filePath = Path.Combine(AppDomain.CurrentDomain.BaseDirectory, relativePath);
+
+
+            _bookings = new List<Booking>();
+            _flightsService = new FlightService();
+
+            if (File.Exists(_filePath))
+            {
+                LoadBookings();
+            }
+        }
+
+        public void AddBooking(Booking booking)
+        {
+            _bookings.Add(booking);
+            SaveBookings();
+        }
+
+        public List<Booking> GetBookingsByPassenger(int passengerId)
+        {
+            return _bookings.Where(b => b.Passenger.Id == passengerId).ToList(); 
+        }
+
+        public Booking GetBookingById(int bookingId)
+        {
+            return _bookings.FirstOrDefault(b => b.Id == bookingId);
+        }
+
+        public void CancelBooking(int bookingId)
+        {
+            var booking = GetBookingById(bookingId);
+            if (booking == null)
+            {
+                Console.WriteLine($"Booking ID {bookingId} Not Found.");
```

Comment:
the right way to handle errors is to throw exceptions not print to screen. the output/printing to screen should be the responsbility for the caller. the repo should just tell the caller you have this exception deal with it the way you want. The caller can either print to the UI, log it, ...